pkg/hub/localizer/localizer.go

```diff
@@ -99,7 +112,282 @@ func (l *Localizer) handleGlobalization(glob *appsapi.Globalization) error {
+					}
+				}
+			}
+
+			outputObjects = append(outputObjects, outputObject)
+		}
+	} else if desc.Spec.Deployer == appsapi.DescriptionGenericDeployer {
+		rawObjects := desc.Spec.Raw
+		for _, object := range rawObjects {
+			outputObject := make([]byte, len(object))
+			copy(outputObject, object)
+
+			unstructured := &unstructured.Unstructured{}
+			err := unstructured.UnmarshalJSON(object)
+			if err != nil {
+				msg := fmt.Sprintf("failed to unmarshal Description %s: %v", klog.KObj(desc), err)
+				klog.Error(msg)
+				l.recorder.Event(desc, corev1.EventTypeWarning, "FailedMarshalingDescription", msg)
+				return err
+			}
+
+			globalizations, err := l.getGlobalizationByResource(unstructured)
+			if err != nil {
+				msg := fmt.Sprintf("failed to get Globalizations matching for Description %s: %v", klog.KObj(desc), err)
+				klog.Error(msg)
+				l.recorder.Event(desc, corev1.EventTypeNormal, "FailedRetrievingGlobalization", msg)
+				return err
+			}
 
+			if len(globalizations) <= 0 {
+				klog.V(4).Info("cannot found Globalizations matching for Description %s ", klog.KObj(desc))
+			} else {
+				for _, glo := range globalizations {
+					outputObject, err = l.executionOverride(appsapi.GenericOverride, glo.Spec.Overrides, outputObject)
+					if err != nil {
+						msg := fmt.Sprintf("failed to override for Description %s and Globalization %s: %v", klog.KObj(desc), klog.KObj(glo), err)
+						klog.Error(msg)
+						l.recorder.Event(desc, corev1.EventTypeWarning, "FailedGlobalizationObject", msg)
+						return err
+					}
+				}
+			}
+
+			localizations, err := l.getLocalizationByResource(desc.Namespace, unstructured)
+			if err != nil {
+				msg := fmt.Sprintf("failed to get Localizations matching for Description %s: %v", klog.KObj(desc), err)
+				klog.Error(msg)
+				l.recorder.Event(desc, corev1.EventTypeNormal, "FailedRetrievingLocalization", msg)
+				return err
+			}
+
+			if len(localizations) <= 0 {
+				klog.V(4).Info("cannot found Localizations matching for Description %s", klog.KObj(desc))
+			} else {
+				for _, loc := range localizations {
+					outputObject, err = l.executionOverride(appsapi.GenericOverride, loc.Spec.Overrides, outputObject)
+					if err != nil {
+						msg := fmt.Sprintf("failed to override for Description %s and Localization %s: %v", klog.KObj(desc), klog.KObj(loc), err)
+						klog.Error(msg)
+						l.recorder.Event(desc, corev1.EventTypeWarning, "FailedLocalizationObject", msg)
+						return err
+					}
+				}
+			}
+
+			outputObjects = append(outputObjects, outputObject)
+		}
+	}
+
+	desc.Spec.Raw = outputObjects
 	return nil
 }
+
+func (l *Localizer) getGlobalizationByResource(unstructured *unstructured.Unstructured) ([]*appsapi.Globalization, error) {
+	var glos []*appsapi.Globalization
+
+	globalizations, err := l.globLister.List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, glo := range globalizations {
+		if IsFeedMatchObject(glo.Spec.Feed, unstructured) {
```

Comment:
use functions in pkg/utils/feed.go

Reply:
use the label selector to implement the getGlobalizationByxxx function